core/optic/shared/src/main/scala/com/useoptic/diff/shapes/resolvers/ShapesResolvers.scala

```diff
@@ -0,0 +1,42 @@
+package com.useoptic.diff.shapes.resolvers
+
+import com.useoptic.contexts.shapes.Commands._
+import com.useoptic.contexts.shapes._
+import com.useoptic.contexts.shapes.ShapesHelper.CoreShapeKind
+import com.useoptic.diff.shapes.resolvers.ShapesResolvers._
+import com.useoptic.diff.shapes._
+
+object ShapesResolvers {
+  case class ChoiceOutput(parentTrail: ShapeTrail, additionalComponents: Seq[ShapeTrailPathComponent], shapeId: ShapeId, coreShapeKind: CoreShapeKind, bindings: ParameterBindings) {
+    def shapeTrail(): ShapeTrail = parentTrail.withChildren(additionalComponents: _*)
+  }
+
+  case class ResolvedTrail(shapeEntity: ShapeEntity, coreShapeKind: CoreShapeKind, bindings: ParameterBindings)
+  type ParameterBindings = Map[ShapeParameterId, Option[ProviderDescriptor]]
+}
+
+trait ShapesResolvers {
```

Comment:
We should have enforced this more strictly earlier, but we should stop accessing RfcState/ShapesState/RequestsState directly, and only access them through some additional query layer, like ShapesResolvers. Most of the places where I did not remove shapesState from the arguments are only using some simple queries like shapesState.shape() and shapesState.field() so it should be easy to move those into ShapesResolvers and remove ShapesState from the arguments. I'm imagining it should be easy to change ShapesResolvers into SpecResolvers which contains shapesResolvers and requestsResolvers, etc. when the time comes

Reply:
Agreed. That seems reasonable and correct to me, but probably better for a cool-down period, not a step on the critical path.